File: solves/Summation_as_a_service/Summation_as_a_service_solve.c
#include <stdio.h>

int main(int argc, char* argv[])
{
	if (argc!=2) {
		perror("usage: summation <inputfile>");
		return(-1);
	}
	FILE *file;
	char str[256];
	file = fopen(argv[1],"r");
	if (file == NULL) {
		perror("Error opening file");
		return(-1);
	}	
	while (fgets(str,sizeof str,file) != NULL) {
		int sum = 0;
		int i, x, y, min, max;
		sscanf(str, "%d %d", &x, &y);
		if (x > y) {
			max = x;
			min = y;
		}
		else {
			max = y;
			min = x;
		}
		for (i=min; i<=max; i++) {
			sum += i;
		}
		printf("%d\n",sum);
	}
	fclose(file);
	return(0);

}

File: solves/Summation_as_a_service/Summation_as_a_service_solve.py
#!/usr/bin/env python
import sys

usage = "%s <input_file>" % sys.argv[0]

if len(sys.argv) < 2:
    exit(usage)

with open(sys.argv[1],'r') as myinput:
    for line in myinput:
        x, y = [int(i) for i in line.strip().split(' ')]
        high = max(x,y)
        low = min(x,y)
        print sum(range(low,high+1))
